lib/brains/views/customer.py

```diff
@@ -22,12 +23,27 @@ def create(request):
         return Response(form.errors, status=400)
 
     uuid = form.cleaned_data['uuid']
+
+    # Create the solitude objects.
+    buyer, buyer_created = Buyer.objects.get_or_create(uuid=uuid)
+    log.info('Buyer {0}: {1}'
+             .format('created' if buyer_created else 'exists', buyer.pk))
+
+    braintree, braintree_buyer_created = (
+        BraintreeBuyer.objects.get_or_create(buyer=buyer))
+    log.info('BraintreeBuyer {0}: {1}'
+             .format('created' if braintree_buyer_created else 'exists',
+                     braintree.pk))
```

Comment:
I don't like how the BraintreeBuyer object is saved to the database before one is created using the braintree API. This means that the API call fails, the database will be out of sync (I don't see any db rollback here). What benefit does this approach give us? Could you just wait until after the API call to save to the database? Otherwise, I think we'd need a new model field like ``exists_in_api=BooleanField()`` so that we can do lookups. That might make things more complicated though.